src/elf_loader.cpp

```
@@ -642,14 +641,16 @@ class ProgramReader {
     /// - TYPE_SIZE: Replace with sizeof(type)
     ///
-    /// @param prog Program containing the instruction to relocate
+    /// @param prog RawProgram containing the instruction to relocate
     /// @param relo CO-RE relocation descriptor (from .BTF.ext section)
     /// @param btf_data BTF type information for offset calculations
     /// @throws UnmarshalError if relocation is invalid or unsupported
     void apply_core_relocation(RawProgram& prog, const bpf_core_relo& relo,
                                const libbtf::btf_type_data& btf_data) const;
     void process_core_relocations(const libbtf::btf_type_data& btf_data);
 
+    int32_t compute_lddw_reloc_offset_imm(ELFIO::Elf_Sxword addend, ELFIO::Elf_Word index, std::reference_wrapper<EbpfInst> lo_inst) const;
```

Clang (14.0.6)

[warning] function 'compute_lddw_reloc_offset_imm' should be marked [[nodiscard]]

(modernize-use-nodiscard)